pkg/controller/standard.oam.dev/v1alpha1/rollout/rollout_controller.go

```diff
@@ -116,6 +116,18 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
+	rolloutPlanSpec := rollout.Spec.RolloutPlan.DeepCopy()
+	// this is a scale operation, if user don't fill rolloutBatches, fill it with default value
+	if len(h.sourceRevName) == 0 && len(rolloutPlanSpec.RolloutBatches) == 0 {
+		// logic reach here means cannot get an error, so ignore it
+		replicas, _ := getWorkloadReplicasNum(*h.targetWorkload)
+		rolloutPlanSpec.RolloutBatches = []v1alpha1.RolloutBatch{{
```

Comment:
This way maybe cannot solve this problem. This this pr  has failed some test indeed. Please have a look.

Reply:
thx，i will try another way